Sets default path to 7z if P7ZIP_HOME_DIR is not defined.  This patch
is there since p7zip 4.55 and now is only remade for the latest p7zip
version.

The patch is Solaris specific.

--- p7zip_16.02/CPP/myWindows/mySplitCommandLine.cpp.orig	2017-12-06 04:47:23.326449773 -0800
+++ p7zip_16.02/CPP/myWindows/mySplitCommandLine.cpp	2017-12-06 04:55:35.467603914 -0800
@@ -106,9 +106,11 @@
 
   { // define P7ZIP_HOME_DIR
     static char p7zip_home_dir[MAX_PATH];
-    AString dir,name;
-    my_windows_split_path(arguments[0],dir,name);
-    snprintf(p7zip_home_dir,sizeof(p7zip_home_dir),"P7ZIP_HOME_DIR=%s/",(const char *)dir);
+    static char default_path[] = "/usr/lib/7z";
+    char *env;
+    if ((env = getenv("P7ZIP_HOME_DIR")) == NULL)
+      env = default_path;
+    snprintf(p7zip_home_dir,sizeof(p7zip_home_dir),"P7ZIP_HOME_DIR=%s/",env);
     p7zip_home_dir[sizeof(p7zip_home_dir)-1] = 0;
     putenv(p7zip_home_dir);
   }
